packages/kit/src/runtime/client/client.js

```diff
@@ -759,6 +756,8 @@ export function create_client({ target, base, trailing_slash }) {
 				throw server_data_node;
 			}
 
+			const prev_server_data_node = (!changed?.url && previous?.server) || null;
```

Comment:
I realize that `changed.url` is not the perfect condition, because we will want to reuse the server data of a layout load even if the url changes, but I'm not familiar enough with the code base to determine the correct condition.